rules/legacy_xcodeproj.bzl

```diff
@@ -871,7 +871,30 @@ def _populate_xcodeproj_targets_and_schemes(ctx, targets, src_dot_dots, all_tran
         test_host_appname = getattr(target_info, "test_host_appname", None)
         if test_host_appname:
             target_dependencies.append({"target": test_host_appname})
-            target_settings["TEST_HOST"] = "$(BUILT_PRODUCTS_DIR)/{test_host_appname}.app/{test_host_appname}".format(test_host_appname = test_host_appname)
+
+            # TEST_HOST setting for UI tests cannot be set because TEST_HOST conflicts with USES_XCTRUNNER
+            if product_type != "bundle.ui-testing":
+                target_settings["TEST_HOST"] = "$(BUILT_PRODUCTS_DIR)/{test_host_appname}.app/{test_host_appname}".format(test_host_appname = test_host_appname)
+            else:
+                target_settings["TEST_TARGET_NAME"] = test_host_appname
+
+        if product_type == "bundle.ui-testing":
+            # Cannot attach LLDB without a signed `*-Runner.app/*-Runner` binary. Xcode won't
+            # sign the UI test runner unless the UI test target itself is also signed. This
+            # intentionally overrides the project-level setting to enable LLDB for UI tests.
```

Comment:
I imagined this was because of the stub toolchain? This might need some other changes to work on device? Looks like a good fix none the less.